js/ftx.js

```diff
@@ -2498,42 +2484,69 @@ module.exports = class ftx extends Exchange {
         const borrowRateHistories = {};
-        for (let i = 0; i < result.length; i++) {
-            const item = result[i];
-            const currency = this.safeCurrencyCode (this.safeString (item, 'coin'));
-            if (!(currency in borrowRateHistories)) {
-                borrowRateHistories[currency] = [];
+        for (let i = 0; i < response.length; i++) {
+            const item = response[i];
+            const code = this.safeCurrencyCode (this.safeString (item, 'coin'));
+            if (codes === undefined || codes.includes (code)) {
+                if (!(code in borrowRateHistories)) {
+                    borrowRateHistories[code] = [];
+                }
+                const lendingRate = this.safeString (item, 'rate');
+                const borrowRate = Precise.stringMul (lendingRate, Precise.stringAdd ('1', spotMarginBorrowRate));
+                const borrowRateStructure = this.extend (this.parseBorrowRate (item), { 'rate': borrowRate });
+                borrowRateHistories[code].push (borrowRateStructure);
             }
-            const datetime = this.safeString (item, 'time');
-            const lendingRate = this.safeString (item, 'rate');
-            borrowRateHistories[currency].push ({
-                'currency': currency,
-                'rate': Precise.stringMul (lendingRate, Precise.stringAdd ('1', spotMarginBorrowRate)),
-                'timestamp': this.parse8601 (datetime),
-                'datetime': datetime,
-                'info': item,
-            });
         }
         const keys = Object.keys (borrowRateHistories);
         for (let i = 0; i < keys.length; i++) {
-            const key = keys[i];
-            borrowRateHistories[key] = this.filterByCurrencySinceLimit (borrowRateHistories[key], key, since, limit);
+            const code = keys[i];
+            borrowRateHistories[code] = this.filterByCurrencySinceLimit (borrowRateHistories[code], code, since, limit);
         }
         return borrowRateHistories;
     }
 
-    async fetchBorrowRateHistory (code, since = undefined, limit = undefined, params = {}) {
-        const histories = await this.fetchBorrowRateHistories (since, limit, params);
-        const borrowRateHistory = this.safeValue (histories, code);
-        if (borrowRateHistory === undefined) {
-            throw new BadRequest (this.id + '.fetchBorrowRateHistory returned no data for ' + code);
-        } else {
-            return borrowRateHistory;
+    parseBorrowRates (response, codeKey) {
+        const result = {};
+        for (let i = 0; i < response.length; i++) {
+            const item = response[i];
+            const currency = this.safeString (item, codeKey);
+            const code = this.safeCurrencyCode (currency);
+            const borrowRate = this.parseBorrowRate (item);
+            result[code] = borrowRate;
         }
+        return result;
+    }
+
+    parseBorrowRate (info) {
```

Comment:
I would suggest adding a second argument `currency = undefined` to parseBorrowRate and use it with safeCurrency to keep it uniform with the rest of the parsers (transactions, orders, trades, etc). What do you say?